### PR TITLE
Use platform APIs to get username instead of env variables.

### PR DESCRIPTION
**Pull request by PawanKartikS**
Previously, username was obtained by reading the value of env variables. This PR implements the usage of platform based APIs to get the username.

### CODE REVIEW

**Comment by PawanKartikS** [comment]
I signed it!

**Comment by PawanKartikS** [comment]
Travil check fail: Travis seems to be including `Windows.h` on Linux platforms when building with `CMake wine stable`.

File: src/utilities.cc [hunk 1]
```diff
@@ -296,13 +300,17 @@ const string& MyUserName() {
   return g_my_user_name;
 }
 static void MyUserNameInitializer() {
-  // TODO(hamaji): Probably this is not portable.
+  int err = 0;
+  char user[1024];
 #if defined(OS_WINDOWS)
-  const char* user = getenv("USERNAME");
+  DWORD bufsize = sizeof(user);
```
Comment (sergiud) [inline src/utilities.cc]: This does not look right.

Comment (sergiud) [reply src/utilities.cc]: `user` could be `NULL`.

Comment (PawanKartikS) [reply src/utilities.cc]: I apologize for the delay.
Please look at the complete diff.

`user` was originally of type `const char*`.  I made changes so it is now a string (as required by the APIs) - `char user[1024];`.

So `sizeof(user)` now actually returns `sizeof(char) * 1024`.

**Pull request closed without merging**